truss/templates/control/control/endpoints.py

```diff
@@ -151,3 +152,36 @@ def _is_streaming_response(resp) -> bool:
         if header_name.lower() == "transfer-encoding" and value.lower() == "chunked":
             return True
     return False
+
+
+def _reroute_if_health_check(path: str) -> str:
+    if path == "/v1/models/model":
+        # Reroute health checks to the inference server's /v1/models/model/loaded endpoint
+        path = "/v1/models/model/loaded"
+    return path
+
+
+def _custom_wait_strategy(retry_state: RetryCallState) -> int:
+    # Apply exponential backoff only for ModelNotReady
+    if retry_state.outcome is not None and isinstance(
+        retry_state.outcome.exception(), ModelNotReady
+    ):
+        return min(2**retry_state.attempt_number, 10)
+    # Use fixed wait for other exception types
+    return 1
+
+
+def _custom_stop_strategy(retry_state: RetryCallState) -> bool:
+    # Stop after 10 attempts for ModelNotReady
+    if retry_state.outcome is not None and isinstance(
+        retry_state.outcome.exception(), ModelNotReady
+    ):
+        # Check if the retry limit for ModelNotReady has been reached
+        return retry_state.attempt_number >= 10
+    # For all other exceptions, stop after INFERENCE_SERVER_START_WAIT_SECS
+    seconds_since_start = (
+        retry_state.seconds_since_start
+        if retry_state.seconds_since_start is not None
+        else 0.0
+    )
+    return seconds_since_start >= INFERENCE_SERVER_START_WAIT_SECS
```

Comment:
nit: It shouldn't if I understand the code correctly, but want to double check: does this change the retry behaviour? 

Is it possible for `retry_state.seconds_since_start` to never be set and for this to retry forever?

Reply:
I don't think its possible for `seconds_since_start` to never be set, looks like its set in the `__init__` of `RetryCallState` https://github.com/jd/tenacity/blob/main/tenacity/__init__.py#L532